scipy/io/matlab/mio.py

```diff
@@ -14,30 +14,43 @@
 __all__ = ['mat_reader_factory', 'loadmat', 'savemat', 'whosmat']
 
 
-def _open_file(file_like, appendmat):
+@contextmanager
+def _open_file_context(file_like, appendmat, mode='rb'):
+    f, opened = _open_file(file_like, appendmat, mode)
+    yield f
+    if opened:
+        f.close()
+
+
+def _open_file(file_like, appendmat, mode='rb'):
     """
     Open `file_like` and return as file-like object. First, check if object is
     already file-like; if so, return it as-is. Otherwise, try to pass it
     to open(). If that fails, and `file_like` is a string, and `appendmat` is true,
     append '.mat' and try again.
     """
     try:
-        file_like.read(0)
-        return file_like, False
+        if hasattr(file_like, 'read') and hasattr(file_like, 'write'):
+            # can't use file_like.read(0) to see if something is a file-like
+            # this is now being used to open a file for writing ('wb') and
+            # the resulting file may be write only.
+            # file_like.read(0)
+            return file_like, False
     except AttributeError:
```

Comment:
Do you need the AttributeError clause now?

Reply:
No, I'll remove it.